junit-jupiter-params/src/main/java/org/junit/jupiter/params/converter/TypedArgumentConverter.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.params.converter;
+
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+
+import org.apiguardian.api.API;
+import org.junit.jupiter.api.extension.ParameterContext;
+
+/**
+ * {@code TypedArgumentConverter} is an {@code ArgumentConverter} that
+ * always converts a given type to another.
+ *
+ * @param <S> the type of the argument to convert
+ * @param <T> the type of the target
+ * @since 5.7
+ * @see ArgumentConverter
+ */
+@API(status = EXPERIMENTAL, since = "5.7")
+public abstract class TypedArgumentConverter<S, T> implements ArgumentConverter {
+
+	private final Class<S> sourceType;
+	private final Class<T> targetType;
+
+	protected TypedArgumentConverter(Class<S> sourceType, Class<T> targetType) {
```

Comment:
Please take a look at `TypeBasedParameterResolver` which determines the parameter type reflectively which would avoid the boilerplate constructor.

Reply:
Thing is, I would have to cast the found `Class` objects into `Class<S>` and `Class<T>`? Or am I missing something?

Reply:
I guess so... 🤔 @sbrannen WDYT?